packages/atomic/src/components/atomic-result-template/atomic-result-template.tsx

```diff
@@ -23,8 +26,30 @@ export class AtomicResultTemplate {
 
   @State() private error?: Error;
 
+  /**
+   * Functions that must return true on results for the result template to apply.
+   *
+   * For example, a template with the following only applies to results whose `title` contains `singapore`:
+   * `[(result) => /singapore/i.test(result.title)]`
+   */
   @Prop() public conditions: ResultTemplateCondition[] = [];
+
+  /**
+   * Fields and field values that results must match for the result template to apply.
+   *
+   * For example, a template with the following attribute only applies
+   * to result items whose `filetype` is `lithiummessage` or `YouTubePlaylist`:
+   * `must-match-filetype="lithiummessage,YouTubePlaylist"`
+   */
   @MapProp() public mustMatch: Record<string, string[]> = {};
```

Comment:
@ThibodeauJF you mentioned that you created this MapProp and after looking into it, it seems we'll have to implement something custom to extract the annotations here and other similar cases.
Ping me if you'd like to discuss. For now at least we'll have the annotations in the code.

Reply:
Description looks good to me! Yes those will take a special treatment 😄